#! /usr/bin/env python

# RosInterface configuration for Dynamic reconfigure

PACKAGE='rostful_node'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
#       Name       Type      Level Description     Default Min   Max
#topics = gen.add_group("Topics")
#size_enum = gen.enum([ gen.const("_turtle_pose",      str_t, "/turtle1/pose", "pose"),
#                       gen.const("_turtle_cmd_vel",     str_t, "/turtle1/cmd_vel", "cmd_vel")],
#                     "An enum to set topic to expose")
#
#topics.add("t", str_t, 0, "A topic parameter which is edited via an enum", "", 0, 1, edit_method=size_enum)

gen.add("topics", str_t, 0, "Topics", "['/turtle1/pose', '/turtle1/cmd_vel']")
gen.add("services", str_t, 0, "Services", "[]")
gen.add("actions", str_t, 0, "Actions", "[]")
gen.add("enable_rocon", bool_t, 0, "Enable Rocon", False)
gen.add("rapps_namespaces", str_t, 0, "Rapps Namespaces", "[]")
gen.add("interactions", str_t, 0, "Interactions", "[]")

exit(gen.generate(PACKAGE, "rostful_node", "RostfulNode"))

